src/DataReader/component_loader.rs

```diff
@@ -7,15 +9,31 @@ use crate::ModelObjects::system_declarations::SystemDeclarations;
 use crate::ProtobufServer::services::query_request::Settings;
 use crate::System::input_enabler;
 use std::collections::HashMap;
-use std::sync::{Arc, RwLock};
+use std::num::NonZeroUsize;
+use std::sync::{Arc, Mutex};
 
 type ComponentsMap = HashMap<String, Component>;
 
+struct ComponentTuple {
+    components_hash: u32,
+    components_map: Arc<ComponentsMap>,
+}
+
 /// A struct used for caching the models.
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Clone)]
 pub struct ModelCache {
-    // TODO: A concurrent hashmap may be faster to use and cause less prone to locking, but is not part of the standard library.
-    cache: Arc<RwLock<HashMap<u32, Arc<ComponentsMap>>>>,
+    // TODO: A concurrent lru may be faster to use and cause less prone to lock contention.
```

Comment:
Noget i vil tilføje nu, @t-lohse plejer ikke være fan af at merge todoes ind i main

Reply:
Nej det var mere for at sige at det er måske muligt at improve performance endnu mere